### PR TITLE
Make deployment of the S3 credentials endpoint optional

### PR DESCRIPTION
**Pull request by yjpa7145**
Users deploying outside of NGAP will not have the NGAP S3 credentials Lambda function, which is required by the S3 credentials endpoint. This PR makes deployment of that endpoint optional, defaulting to `true` (the previous behavior).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 **Please note**:
 - Your workflow tasks should use `cumulus-message-adapter-js` version 1.0.10+ to utilize new granule, parentArn, asyncOperationId, and stackName fields on the logs.
 
+### BREAKING CHANGES
+
+- **CUMULUS-1321**
+  - The `region` argument to the `cumulus` Terraform module has been removed
```
Comment (yjpa7145) [inline CHANGELOG.md]: This was only used in the `archive` module, and is not longer used in that module.

File: example/cumulus-tf/main.tf [hunk 2]
```diff
@@ -40,7 +40,6 @@ module "cumulus" {
   cumulus_message_adapter_lambda_layer_arn = var.cumulus_message_adapter_lambda_layer_arn
 
   prefix = var.prefix
-  region = var.region
```
Comment (yjpa7145) [inline example/cumulus-tf/main.tf]: This was removed from the `cumulus` module.

File: packages/api/app/distribution-routes.js [hunk 3]
```diff
@@ -101,7 +100,6 @@ async function ensureAuthorizedOrRedirect(req, res, next) {
 }
 
 router.get('/redirect', handleRedirectRequest);
-router.get('/s3credentials', ensureAuthorizedOrRedirect, handleCredentialRequest);
```
Comment (yjpa7145) [inline packages/api/app/distribution-routes.js]: The `s3Credentials` code in the `api` package is no longer used and has been removed.

File: tf-modules/archive/api.tf [hunk 4]
```diff
@@ -44,7 +44,6 @@ resource "aws_lambda_function" "api" {
       ProvidersTable               = var.dynamo_tables.providers.name
       RulesTable                   = var.dynamo_tables.rules.name
       oauth_user_group             = var.oauth_user_group
-      STSCredentialsLambda         = var.sts_credentials_lambda
```
Comment (yjpa7145) [inline tf-modules/archive/api.tf]: This is not used and should have been removed a while back.

File: packages/api/app/distribution-routes.js (hunk 3 above)
Comment (laurenfrederick) [reply packages/api/app/distribution-routes.js]: Nice catch

**Comment by yjpa7145** [comment]
> The `sts_credentials_lambda_function_arn` doesn't seem to be optional? I think if we have the option to not deploy, this should be optional.

Fixed in e137fe31a

**Comment by yjpa7145** [comment]
> Did you update the `template-deploy` repo to include this with some notes?

https://github.com/nasa/cumulus-template-deploy/pull/25

**Pull request merged**